app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/activities/SearchActivity.java

```diff
@@ -0,0 +1,179 @@
+package com.example.edwinmperazaduran.gridimagesearch.activities;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.Bundle;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatActivity;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.AdapterView;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.GridView;
+import android.widget.Toast;
+
+import com.example.edwinmperazaduran.gridimagesearch.Dialogs.ImageFilterDialog;
+import com.example.edwinmperazaduran.gridimagesearch.R;
+import com.example.edwinmperazaduran.gridimagesearch.adapters.ImageResultArrayAdapter;
+import com.example.edwinmperazaduran.gridimagesearch.helpers.EndlessScrollListener;
+import com.example.edwinmperazaduran.gridimagesearch.models.ImageFilter;
+import com.example.edwinmperazaduran.gridimagesearch.models.ImageResult;
+import com.example.edwinmperazaduran.gridimagesearch.net.SearchClient;
+import com.loopj.android.http.JsonHttpResponseHandler;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+import cz.msebera.android.httpclient.Header;
+
+public class SearchActivity extends AppCompatActivity implements
+        ImageFilterDialog.ImageFilterDialogListener{
+
+    private static int MAX_PAGE = 8;
+    EditText etQuery;
```

Comment:
@edwinperaza you should be explicit on the `public` or `private` matter of the variable.